src/hotspot/share/classfile/systemDictionary.cpp

```diff
@@ -1873,9 +1873,14 @@ void SystemDictionary::add_nest_host_error(const constantPoolHandle& pool,
       // An existing entry means we had a true resolution failure (LinkageError) with our nest host, but we
       // still want to add the error message for the higher-level access checks to report. We should
       // only reach here under the same error condition, so we can ignore the potential race with setting
-      // the message, and set it again.
-      assert(entry->nest_host_error() == nullptr || strcmp(entry->nest_host_error(), message) == 0, "should be the same message");
-      entry->set_nest_host_error(message);
+      // the message.
+      const char* nhe = entry->nest_host_error();
+      if (nhe == nullptr) {
+        entry->set_nest_host_error(message);
+      }
+      else {
+        assert(strcmp(nhe, message) == 0, "New message %s, differs from original %s", message, nhe);
+      }
```

Comment:
The caller allocates this message so the caller should free it but that would require passing up that we actually installed the message.  We could keep this leak temporarily until https://bugs.openjdk.org/browse/JDK-8372373 is fixed to resource allocate the message.  Or add a FREE_C_HEAP_ARRAY here even though it's not the best place for it?